File: 1672.richest-customer-wealth.cpp
#include <numeric>
#include <vector>
using namespace std;

class Solution {
 public:
  int maximumWealth(vector<vector<int>> &accounts) {
    int max_wealth = 0;
    for (auto &customer : accounts)
      max_wealth =
        max(max_wealth, accumulate(customer.begin(), customer.end(), 0));
    return max_wealth;
  }
};
int main() {
  Solution sol;
  vector<vector<int>> accounts = {{1, 2, 3}, {3, 2, 1}};
  assert(6 == sol.maximumWealth(accounts));
  return 0;
}

File: 1672.richest-customer-wealth.go
func maximumWealth(accounts[][] int) int {
  max_wealth := 0;
  for i := range accounts {
    curr_wealth := 0;
    for j := range accounts[i] {
      curr_wealth += accounts[i][j];
    }
    if max_wealth < curr_wealth {
      max_wealth = curr_wealth;
    }
  }
  return max_wealth
}
